protocol/deposits-interop.md

```diff
@@ -0,0 +1,189 @@
+race conditions with deposit transactions and the sequencer having a slow p2p network. Without this invariant,
+it would be possible to trigger reorgs by sending a deposit transaction that references an initiating message
+that is at the tip of a remote chain. If the sequencer doesn't immediately always have the remote chain, it
+could result in reorgs.
+
+The definition of the sequencing window invariant is as follows:
+
+```
+require(identifier.timestamp + sequencer_window <= block.timestamp)
+```
+
+Where the `block.timestamp` is the timestamp of the pending block, ie the next block that is being built.
+
+This states that a sequencing window must bass on top of an initiating message before it can be considered
+valid via a deposit. This removes the concept of p2p based preconfirmations from consensus. If the sequencing
+window has elapsed on top of an initiating message, it can be safely considered as final.
+
+# Alternatives Considered
+
+The design spaces for solutions involve one of the following patterns:
+- guarantee that there is no way to create an invalid executing message with a deposit
+- update the mapping from events to deposit transactions so that it can result in noop deposit transactions
+
+## Banning Deposits Creating Executing Messages
+
+With the "only EOA invariant", this could be handled at the level of the `OptimismPortal`. The idea
+is that the protocol itself could ban deposits that call the `CrossL2inbox` via static analysis.
+This design is not possible with the removal of the "only EOA invariant". An arbitrary subcall
+can call out to the `CrossL2Inbox`.
+
+With the "only EOA invariant", this could also be handled in the derivation pipeline. The `DepositTx`
+could be mapped into a null value if its `tx.to` as equal to the `address(CrossL2Inbox)`.
+
+## State Transition Diff
+
+It would be possible to not require that there is execution in the derivation pipeline but this would
+result in the requirement for the EL to malleate `ExecutionPayload`s. The execution payload contains
+the raw RLP hex serialized transations for a block. Right now there is an invariant such that the
+`ExecutionPayload` is just accepted as is and is executed. It includes the deposit transactions
+and the `op-node` is responsible for building the `ExecutionPayload` correctly. It is possible to
+have the EL modify (malleate) the `ExecutionPayload`'s deposit transactions before considering
+them final. This would essentially be adding more rules to the state transition function.
+
+This would look like a diff to the state transition function that applies the sequencing
+window invariant and does the executing message validity checking during the state transition.
+This creates a leaky abstraction as now the state transition function needs to know about
+the dependency set, doing remote RPC calls during execution. This does not seem like
+an ideal solution for that reason.
+
```

Comment:
This would also impact p2p networking as the execution payloads are sent via p2p, would need to make sure to define semantics around signing the payload before/after the malleation